### PR TITLE
[New] Add monaco-editor@0.9.0 w/ npm auto-update

### PR DESCRIPTION
**Pull request by sufuf3**
Pull request for issue: #9351
Related issue(s): #9369 #11170

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/Microsoft/monaco-editor
 * Official website (optional, not the repository): https://microsoft.github.io/monaco-editor/
 * NPM package url (optional): https://www.npmjs.com/package/monaco-editor
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 202
   - Count of stars: 4075
   - Count of forks: 323
 * NPM download stats (optional):
   - Downloads in the last day: 600
   - Downloads in the last week: 3315
   - Downloads in the last month: 12952

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by sufuf3** [comment]
@cdnjs/intern3 
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@cdnjs/intern3
Please help me review this PR, Thank you.

File: ajax/libs/monaco-editor/package.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  "keywords": [
+    "Monaco",
+    "Editor",
+    "TypeScript",
+    "JavaScript"
+  ],
+  "npmName": "monaco-editor",
+  "npmFileMap": [
+    {
+      "basePath": "",
```
Comment (extend1994) [inline ajax/libs/monaco-editor/package.json]: Can we put `min` and `min-maps` in basePath? It't similar to minified I think.

File: ajax/libs/monaco-editor/package.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/Microsoft/monaco-editor.git"
+  },
+  "homepage": "https://microsoft.github.io/monaco-editor/",
+  "keywords": [
+    "Monaco",
+    "Editor",
+    "TypeScript",
+    "JavaScript"
```
Comment (extend1994) [inline ajax/libs/monaco-editor/package.json]: Maybe one more "vscode"? It appeared in GitHub repo keywords area.

**Comment by sufuf3** [comment]
@extend1994 I modified `keywords` and `basePath`
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@cdnjs/intern3 @dakshshah96 
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@cdnjs/library-reviewer 
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@hare1039 @qwe19272375 @pvnr0082t @maruilian11 
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@hare1039 @qwe19272375 @pvnr0082t @maruilian11
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@PeterDaveHello 
Please help me review this PR, Thank you.
I can't require your help to review this PR because I got the message `You can't perform that action at this time.`.

**Comment by sufuf3** [comment]
@maruilian11 I added dev folder, 
Please help me review this PR, Thank you.

File: ajax/libs/monaco-editor/package.json (hunk 1 above)
Comment (PeterDaveHello) [inline ajax/libs/monaco-editor/package.json]: By the latest line of `min/vs/loader.js` which referred to the map file by relative path, we better keep their structure.

**Comment by sufuf3** [comment]
@pvnr0082t @maruilian11 
Please help me review this PR, Thank you.

File: ajax/libs/monaco-editor/package.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    "vscode",
+    "TypeScript",
+    "JavaScript"
+  ],
+  "npmName": "monaco-editor",
+  "npmFileMap": [
+    {
+      "basePath": "",
+      "files": [
+        "+(dev|min|min-maps)/**/!(*.json)"
```
Comment (maruilian11) [inline ajax/libs/monaco-editor/package.json]: Since it's not a dist folder, l think use `+(dev|min|min-maps)/**/*.+(js|css|map|svg)` will be better.

**Comment by sufuf3** [comment]
@maruilian11 I modified the `filemap` rule
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@hare1039 @qwe19272375 @pvnr0082t @maruilian11
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@dakshshah96 @qwe19272375 @extend1994 @maruilian11 @pvnr0082t 
Please help me review this PR, Thank you.

Memo: 
1. As https://github.com/cdnjs/cdnjs/pull/11321#discussion_r126308836 I keep the file structure as upstream
2. Although https://github.com/cdnjs/cdnjs/pull/11321#pullrequestreview-48391286, https://github.com/cdnjs/cdnjs/pull/11321#discussion_r127667804, maintainer said https://github.com/cdnjs/cdnjs/pull/11321#pullrequestreview-55189772 so I added `min` and `min-maps`.

**Comment by sufuf3** [comment]
@dakshshah96 @pvnr0082t @maruilian11 
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@hare1039 @qwe19272375 @pvnr0082t @maruilian11
Please help me review this PR, Thank you.

**Pull request merged**